### PR TITLE
Publish akka-osgi but do not include it in dist #20548

### LINKED ISSUE #20548: akka-osgi not part of release
After simplifying the build in #20486 the akka-osgi module is no longer published to maven central. 

It should be published but it should not be included in the dist zip.

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #20548

### CODE REVIEW

File: project/Dist.scala [hunk 1]
```diff
@@ -24,8 +24,10 @@ object Dist {
   val distDocJars = TaskKey[Seq[File]]("dist-doc-jars")
   val distSources = TaskKey[DistSources]("dist-sources")
   val dist = TaskKey[File]("dist", "Create a zipped distribution of everything.")
+  val includeInDist = SettingKey[Boolean]("include-in-dist", "Include the artifact of this project in the standalone dist zip-file")
 
   lazy val settings: Seq[Setting[_]] = Seq(
+    includeInDist := true,
```
Comment (johanandren) [inline project/Dist.scala]: so, defaults to true here, and then changes to false in dontPublishSettings, but also allows for akka-osgi to opt out of dist but still be published.

**Comment by johanandren** [comment]
Compared dist output zip to akka_2.11-2.4.5.zip on akka.io, has the exact same size and does not contain akka-osgi jar. Ran `publishSigned` locally and it publishes the jar. Something more I should try out?

**Comment by patriknw** [comment]
LGTM

File: akka-osgi/build.sbt [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import akka.{ AkkaBuild, Dependencies, Formatting, OSGi }
+import akka.{ AkkaBuild, Dependencies, Formatting, OSGi, Dist }
 
 AkkaBuild.defaultSettings
-AkkaBuild.dontPublishSettings
+Dist.includeInDist := false
```
Comment (2beaucoup) [inline akka-osgi/build.sbt]: I know there was the ticket but it didn't explain *why* it shouldn't be part of the dist. Do you know it?

Comment (johanandren) [inline akka-osgi/build.sbt]: Could it be that it duplicates stuff? I'd rather make it behave the same and have someone convince me to change it than to change it without knowing though :)

Comment (patriknw) [inline akka-osgi/build.sbt]: I'm not sure, but I don't osgi can be used without maven repo, so it has no value in the dist

**Comment by ktoso** [comment]
LGTM

File: project/Dist.scala (hunk 1 above)
Comment (2beaucoup) [inline project/Dist.scala]: so this was picked up just by `root`?

Comment (johanandren) [inline project/Dist.scala]: Yeah, was part of `rootSettings` which I incorrectly assumed was inherited by the modules

**Pull request merged**